pkg/subctl/cmd/show_endpoints.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	submarinerclientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type endpointStatus struct {
+	clusterId    string
+	endpointIp   string
+	publicIp     string
+	cableDriver  string
+	endpointType string
+}
+
+func newEndpointsStatusFrom(clusterId string, endpointIp string, publicIp string, cableDriver string, endpointType string) endpointStatus {
+	return endpointStatus{
+		clusterId:    clusterId,
+		endpointIp:   endpointIp,
+		publicIp:     publicIp,
+		cableDriver:  cableDriver,
+		endpointType: endpointType,
+	}
+}
+
+var showEndpointsCmd = &cobra.Command{
+	Use:   "endpoints",
+	Short: "Show submariner endpoint information",
+	Long:  `This command shows information about submariner endpoints in a cluster.`,
+	Run:   showEndpoints,
+}
+
+func init() {
+	showCmd.AddCommand(showEndpointsCmd)
+}
+
+func getEndpointsStatus(status []endpointStatus) []endpointStatus {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config for cluster", err)
+
+	submarinerClient, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner client", err)
+
+	existingCfg, err := submarinerClient.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, v1.GetOptions{})
+	if err != nil {
+		exitOnError("Error obtaining the Submariner resource", err)
+	}
+
+	gateways := existingCfg.Status.Gateways
+	if gateways == nil {
+		exitWithErrorMsg("No endpoints found")
+	}
+
+	for _, gateway := range *gateways {
+		status = append(status, newEndpointsStatusFrom(
+			gateway.Status.LocalEndpoint.ClusterID,
+			gateway.Status.LocalEndpoint.PrivateIP,
+			gateway.Status.LocalEndpoint.PublicIP,
+			gateway.Status.LocalEndpoint.Backend,
+			"local"))
+
+		for _, connection := range gateway.Status.Connections {
+			status = append(status, newEndpointsStatusFrom(
+				connection.Endpoint.ClusterID,
+				connection.Endpoint.PrivateIP,
+				connection.Endpoint.PublicIP,
+				connection.Endpoint.Backend,
+				"remote"))
+		}
+	}
+
+	return status
+}
+
+func showEndpoints(cmd *cobra.Command, args []string) {
+	var status []endpointStatus
+	status = getEndpointsStatus(status)
```

Comment:
```suggestion
	status = getEndpointsStatus()
```

Reply:
This is more natural this way because it isn't really an input to the function, the function won't consume that data, so it doesn't make sense to pass it down.